docs/reference/local_development.md

```diff
@@ -0,0 +1,198 @@
+# Local Development
+
+*This document is a conceptual overview of how Tye behaves when using `tye run` for local development. For reference docs on the `tye run` command see [here](/docs/reference/commandline/tye-run.md).*
+
+Tye acts as a *local orchestrator* for development by building and launching multiple services, managing and monitoring services, and enabling network communication between services.
+
+This document will describe how these features work, as well as the implications and behaviors of various optional settings.
+
+## Executing `tye run`
+
+When executing `tye run` for local development, Tye goes through the following steps:
+
+- Load the application (`tye.yaml`)
+- Build all projects 
```

Comment:
Should we mention extensions here?

Reply:
I wasn't planning to yet. That seems like a level of detail we don't need.